src/main/java/org/veupathdb/service/eda/compute/plugins/alphadiv/AlphaDivPlugin.java

```diff
@@ -0,0 +1,60 @@
+package org.veupathdb.service.eda.compute.plugins.alphadiv;
+
+import org.gusdb.fgputil.ListBuilder;
+import org.jetbrains.annotations.NotNull;
+import org.veupathdb.service.eda.common.client.spec.StreamSpec;
+import org.veupathdb.service.eda.common.model.VariableDef;
+import org.veupathdb.service.eda.common.plugin.util.PluginUtil;
+import org.veupathdb.service.eda.compute.plugins.AbstractPlugin;
+import org.veupathdb.service.eda.compute.plugins.PluginContext;
+import org.veupathdb.service.eda.compute.RServe;
+import org.veupathdb.service.eda.generated.model.AlphaDivPluginConfig;
+import org.veupathdb.service.eda.generated.model.AlphaDivPluginRequest;
+import org.veupathdb.service.eda.generated.model.VariableSpec;
+
+
+import java.util.List;
+
+public class AlphaDivPlugin extends AbstractPlugin<AlphaDivPluginRequest, AlphaDivPluginConfig> {
+
+  private static final String INPUT_DATA = "alpha_div_input";
+
+  public AlphaDivPlugin(@NotNull PluginContext<AlphaDivPluginRequest, AlphaDivPluginConfig> context) {
+    super(context);
+  }
+
+  @NotNull
+  @Override
+  public List<StreamSpec> getStreamSpecs() {
+    return List.of(new StreamSpec(INPUT_DATA, getConfig().getCollectionVariable().getEntityId())
+        .addVars(getUtil().getChildrenVariables(getConfig().getCollectionVariable())
+      ));
+  }
+
+  @Override
+  protected void execute() {
+    
+    AlphaDivPluginConfig computeConfig = getConfig();
+    PluginUtil util = getUtil();
+    VariableDef computeEntityIdVarSpec = util.getEntityIdVarSpec(computeConfig.getCollectionVariable().getEntityId());
+    String computeEntityIdColName = util.toColNameOrEmpty(computeEntityIdVarSpec);
+    String method = computeConfig.getAlphaDivMethod().getName();
+    
+    RServe.useRConnection(connection -> {
+      connection.voidEval("print('starting alpha diversity computation')");
+
+      List<VariableSpec> computeInputVars = ListBuilder.asList(computeEntityIdVarSpec);
+      computeInputVars.addAll(util.getChildrenVariables(computeConfig.getCollectionVariable()));
+      connection.voidEval(util.getVoidEvalFreadCommand(INPUT_DATA, computeInputVars));
+
+      connection.voidEval("alphaDivDT <- alphaDiv(" + INPUT_DATA + ", " + 
+                                                    util.singleQuote(computeEntityIdColName) + ", " + 
+                                                    util.singleQuote(method) + ")");
+      String dataCmd = "print(alphaDivDT)";
+      String metaCmd = "getMetadata(alphaDivDT)";
```

Comment:
I'm not sure if the alpha div R package exports a getMetaData?
[beta div does](https://github.com/VEuPathDB/microbiomeComputations/blob/574ebd79e6bea6ff25f63342215319c5bd0ce55c/R/app_betaDiv.R#L164)

Reply:
getMetadata is currently written as a generic fxn as opposed to a method specific to a class, the one defined in the beta div file will work for any of the computes currently. This is something I plan to refactor and do properly soon but don't want to hold up a release of beta div for.